contracts/base/escrow/dapps/Aave.sol

```diff
@@ -0,0 +1,135 @@
+    using Address for address;
+    using SafeERC20 for IERC20;
+
+    /**
+        @notice The deposit, the aToken needs to be approved to have access to the token balance
+        @param tokenAddress address of the token
+        @param amount amount of tokens to deposit
+     */
+    function deposit(address tokenAddress, uint256 amount) public onlyBorrower {
+        IAToken aToken = _getAToken(tokenAddress);
+        IAaveLendingPool aaveLendingPool = _getAavePoolAddress(tokenAddress);
+        uint256 aTokenBalanceBeforeDeposit = aToken.balanceOf(address(this));
+        IERC20(tokenAddress).safeApprove(address(aaveLendingPool), amount);
+        aaveLendingPool.deposit(tokenAddress, amount, address(this), 0);
+
+        uint256 aTokenBalanceAfterDeposit = aToken.balanceOf(address(this));
+        require(
+            aTokenBalanceAfterDeposit > aTokenBalanceBeforeDeposit,
+            "AAVE_BALANCE_NOT_INCREASED"
+        );
+
+        _tokenUpdated(address(aToken));
+        _tokenUpdated(tokenAddress);
+
+        uint256 tokenBalanceAfterDeposit = _balanceOf(tokenAddress);
+        emit AaveDeposited(
+            tokenAddress,
+            address(aToken),
+            amount,
+            tokenBalanceAfterDeposit,
+            aTokenBalanceAfterDeposit
+        );
+    }
+
+    /**
+        @notice This function withdraws the user's aTokens for a specific amount
+        @param tokenAddress address of the token
+        @param amount amount of the underlying tokens to withdraw
+     */
+    function withdraw(address tokenAddress, uint256 amount)
+        public
+        onlyBorrower
+    {
+        IAToken aToken = _getAToken(tokenAddress);
+        IAaveLendingPool aaveLendingPool = _getAavePoolAddress(tokenAddress);
+        uint256 aTokenBalanceBeforeWithdraw = aToken.balanceOf(address(this));
+        require(aTokenBalanceBeforeWithdraw >= 0, "NO_BALANCE_TO_WITHDRAW");
+        uint256 result =
+            aaveLendingPool.withdraw(tokenAddress, amount, address(this));
+        uint256 aTokenBalanceAfterWithdraw = aToken.balanceOf(address(this));
+        require(
+            aTokenBalanceAfterWithdraw < aTokenBalanceBeforeWithdraw,
+            "AAVE_WITHDRAWAL_ERROR"
+        );
+
+        _tokenUpdated(address(aToken));
+        _tokenUpdated(tokenAddress);
+
+        uint256 tokenBalanceAfterWithdrawal = _balanceOf(tokenAddress);
+        emit AaveWithdrawn(
+            tokenAddress,
+            address(aToken),
+            amount,
+            tokenBalanceAfterWithdrawal,
+            aTokenBalanceAfterWithdraw
+        );
+    }
+
+    /**
+        @notice Grabs the Aave lending pool instance from the Aave lending pool address provider
+        @param tokenAddress The token address to get the lending pool for
+        @return IAaveLendingPool instance address
+     */
+    function _getAavePoolAddress(address tokenAddress)
+        internal
+        view
+        returns (IAaveLendingPool)
```

Comment:
Function is returning the LP interface, not just the address. Lets change the function name accordingly